monitor/src/main/java/com/iluwatar/monitor/Main.java

```diff
@@ -0,0 +1,57 @@
+package com.iluwatar.monitor;
+
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class Main {
+
+    public static void main(String[] args) {
+        AtomicReference<Bank> bank = new AtomicReference<>();
+        bank.set(new Bank(4, 1000));
+        Runnable runnable = () -> {
+            try {
+                Thread.sleep((long) (Math.random() * 1000));
+                Random random = new Random();
+                for (int i = 0; i < 1000000; i++)
+                    bank.get().transfer(random.nextInt(4), random.nextInt(4), (int) (Math.random() * 1000));
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        };
+        ExecutorService executorService = Executors.newFixedThreadPool(5);
+        executorService.execute(runnable);
+        executorService.execute(runnable);
+        executorService.execute(runnable);
+        executorService.execute(runnable);
+        executorService.execute(runnable);
+    }
+}
+
+//We call this class as monitor !
+class Bank {
+
+    private int[] accounts;
+
+    public Bank(int accountNum, int baseAmount) {
+        accounts = new int[accountNum];
+        Arrays.fill(accounts, baseAmount);
+    }
+
+    public synchronized void transfer(int accountA, int accountB, int amount) {
```

Comment:
Isnt it better to read/write locks instead of synchronized keyword?
It may not make much difference in this example but as the number of thread increases, this will have severe overhead

Reply:
of course ! 
we can use ReadWriteLock in critical sections to implement CREW strategy . but these codes are for sample to understand .

Reply:
As @ravening said, the sample code needs to use the best practices.

Reply:
i know !
be sure these code best practice to implementing monitor .